src/Partials/Components/datetime.jsx

```diff
@@ -10,10 +10,36 @@ module.exports = React.createClass({
   displayName: 'Datetime',
   mixins: [valueMixin, multiMixin],
   getInitialValue: function() {
-    return null;
+    if (
+      typeof this.props.component === "undefined" ||
+      typeof this.props.component.defaultDate === "undefined" ||
+      this.props.component.defaultDate.length === 0
+    ) {
+      return '';
+    }
+    const { defaultDate } = this.props.component;
+
+    var dateVal = new Date(defaultDate);
+    if (isNaN(dateVal.getDate())) {
+      try {
+        dateVal = new Date(eval(defaultDate));
+      }
+      catch (e) {
+        dateVal = '';
+      }
+    }
+
+    if (isNaN(dateVal)) {
+      dateVal = '';
+    }
+
+    return dateVal;
   },
   onChangeDatetime: function(index, value, str) {
-    this.setValue(value, index);
+    this.setValue(
+      this.parseMomentValue(value),
```

Comment:
Where is this.parseMomentValue defined? I'd like to do this the same way we did in angular where the eval statement in line 25 has moment and Date available to be evaled.